packages/api/models/providers.js

```diff
@@ -116,6 +116,24 @@ class Provider extends Manager {
     await super.delete({ id });
   }
 
+
+  /**
+   * Scan all items in a provider table
+   * Note: Only used for tests. Should not be used for large datatables
+   */
+  async scan() {
+    const scanResults = await super.scan();
+    return scanResults.Items;
+  }
+
+  async deleteProviders() {
+    const providers = await this.scan();
+    const providerIds = providers.map((p) => p.id);
```

Comment:
It can be simplified as in `deleteRules()`:
`const deletes = providers.map((p)=>this.delete({id: p.id}));
await Promise.all(deletes)`